Tests/NIOPosixTests/NIOThreadPoolTest.swift

```diff
@@ -110,6 +111,21 @@ class NIOThreadPoolTest: XCTestCase {
         }
     }
 
+    func testAsyncThreadPool() async throws {
+        guard #available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *) else { throw XCTSkip() }
+        let numberOfThreads = 1
+        let pool = NIOThreadPool(numberOfThreads: numberOfThreads)
+        pool.start()
+        do {
+            let hitCount = ManagedAtomic(false)
+            try await pool.runIfActive {
+                hitCount.store(true, ordering: .relaxed)
+            }
+            XCTAssertEqual(hitCount.load(ordering: .relaxed), true)
+        } catch {}
+        try await pool.shutdownGracefully()
+    }
```

Comment:
Can we test the two other paths through the code as well?

Reply:
Do you mean throwing errors from the closure, and the thread pool not being active

Reply:
I do, yeah. 😄